src/agents/quest_agent.py

```diff
@@ -354,9 +354,10 @@ def create_problem(
                 description=updated_problem_block.text, context=context
             )
         if music_gen := get_music_generator(context):
-            music_gen.request_scene_music_generation(
-                description=updated_problem_block.text, context=context
-            )
+            if server_settings.generate_music:
+                music_gen.request_scene_music_generation(
```

Comment:
thought (non-blocking): this makes me wonder about having either `get_music_generator()` return `None` when `server_settings.generate_music` is `False` or having the generator itself no-op on `request_scene_music_generation()` for the same condition. Might clean up the code

Reply:
Yeah I was thinking the exact same thing but worrying about handling the possibly None task made me pick the lazy route -- maybe second cleanup pass task?